### ISSUE TITLE
IDE0045 Bad suggestion results in CS8602 false-positive, revert results in bad IDE0045 suggestion

### ISSUE THREAD

**Issue by AliveDevil** [dotnet/roslyn#75200]
**Version Used**:
```
.NET SDK:
 Version:           8.0.401
 Commit:            811edcc344
 Workload version:  8.0.400-manifests.b6724b7a
 MSBuild version:   17.11.4+37eb419ad
```

**Steps to Reproduce**:

Both these methods produce IDE0045:
```
public void M2(object[]? parent, int i, object value)
{
    if (parent == null) // IDE0045
    {
        throw new KeyNotFoundException();
    }
    parent[i] = value;
}

public void N(object[]? parent, int i, object value)
{
    if (parent is { }) // IDE0045
    {
        parent[i] = value;
    }
    else throw new KeyNotFoundException();
}
```
Which is simplified to 
```
public void M(object[]? parent, int i, object value)
{
    parent[i] = parent is { } ? value : throw new KeyNotFoundException();
}
```
But, this emits analyzer warning CS8602, which on revert round-trips to IDE0045.

[Sharplab](https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA+ABADAAgwRhQG4BYAKHIwGY8AmHAYXIG9yd28aMUcBZACgjAAVjDAAXANoBdAPw4ADgENYAO3EAaHAEt1OrUNEScANyUAbAK4wAlGw6syHZ4pUx1k7dJwBeV2vEdAGccZhwAXxx5MysYHBAccQALKAgAdxxVGAyAaRgATwA5CHEAMQhLVQATAFEEMBgFcW0IVX4bUicOcPt2Xs48Hl5aQRExKTl/d00dPW0DMeMY6zsu9kcXDm0AMxx+ZQDfP1VLc3MbHAB6S5wASQARGqwsFABWfucNzZdk1Iys3IFYplCrVOoNJotNodD7dWHsA7TTzePzLGCdZw9NY4frUQY4QqjIwTeSI9RaXSBeY4QzjUwWFb9L6bHZ7MlUkJhcIXa53R7PN7kACQQscIqF7ORvnpsU6IqxIpg5iCcV+6Uy2RweSKJXKlVq9UazVa7QxcLI4SAA===)

**Diagnostic Id**: 
- IDE0045: 'if' statement can be simplified
- CS8602: Dereference of a possibly null reference.

**Expected Behavior**:
Either,
- IDE0045 is not emitted in the above cases, where CS8602 can't validate nullability guard
- CS8602 is not emitted in the above cases, where IDE0045 validly suggests case simplification.

**Actual Behavior**:
IDE0045 is suggested, but the suggestion results in CS8602, which when reverted suggests IDE0045 again. Repeat.

Disabling CS8602 or IDE0045 is not a valid solution, or viable long term workaround.

Relates: #63441
